pkg/persistence/postgres.go

```diff
@@ -0,0 +1,387 @@
+		CREATE SEQUENCE IF NOT EXISTS ` + db.schema + `.seq_files_id;
+
+		CREATE TABLE IF NOT EXISTS ` + db.schema + `.torrents (
+			id             INTEGER PRIMARY KEY DEFAULT nextval('` + db.schema + `.seq_torrents_id'),
+			info_hash      bytea NOT NULL UNIQUE,
+			name           TEXT NOT NULL,
+			total_size     BIGINT NOT NULL CHECK(total_size > 0),
+			discovered_on  INTEGER NOT NULL CHECK(discovered_on > 0)
+		);
+
+		-- Indexes for search sorting options
+		CREATE INDEX IF NOT EXISTS idx_torrents_total_size ON ` + db.schema + `.torrents (total_size);
+		CREATE INDEX IF NOT EXISTS idx_torrents_discovered_on ON ` + db.schema + `.torrents (discovered_on);
+
+		-- Using pg_trgm GIN index for fast ILIKE queries
+		-- You need to execute "CREATE EXTENSION pg_trgm" on your database for this index to work
+		CREATE INDEX IF NOT EXISTS idx_torrents_name_gin_trgm ON ` + db.schema + `.torrents USING GIN (name gin_trgm_ops);
+
+		CREATE TABLE IF NOT EXISTS ` + db.schema + `.files (
+			id          INTEGER PRIMARY KEY DEFAULT nextval('` + db.schema + `.seq_files_id'),
+			torrent_id  INTEGER REFERENCES ` + db.schema + `.torrents ON DELETE CASCADE ON UPDATE RESTRICT,
+			size        BIGINT NOT NULL,
+			path        TEXT NOT NULL
+		);
+
+		CREATE INDEX IF NOT EXISTS idx_files_torrent_id ON ` + db.schema + `.files (torrent_id);
+
+		CREATE TABLE IF NOT EXISTS ` + db.schema + `.migrations (
+		    schema_version		SMALLINT NOT NULL UNIQUE 
+		);
+
+		INSERT INTO ` + db.schema + `.migrations (schema_version) VALUES (0) ON CONFLICT DO NOTHING;
+	`)
+	if err != nil {
+		return errors.Wrap(err, "sql.Tx.Exec (v0)")
+	}
+
+	// Get current schema version
+	rows, err = tx.Query("SELECT MAX(schema_version) FROM " + db.schema + ".migrations;")
+	if err != nil {
+		return errors.Wrap(err, "sql.Tx.Query (SELECT MAX(version) FROM "+db.schema+".migrations)")
+	}
+	defer db.closeRows(rows)
+
+	var schemaVersion int
+	if rows.Next() != true {
+		return fmt.Errorf("sql.Rows.Next (SELECT MAX(version) FROM " + db.schema + ".migrations): Query did not return any rows")
+	}
+	if err = rows.Scan(&schemaVersion); err != nil {
+		return errors.Wrap(err, "sql.Rows.Scan (MAX(version))")
+	}
+	// If next line is removed we're getting error on sql.Tx.Commit: unexpected command tag SELECT
+	// https://stackoverflow.com/questions/36295883/golang-postgres-commit-unknown-command-error/36866993#36866993
+	db.closeRows(rows)
+
+	// Uncomment for future migrations:
+	//switch schemaVersion {
+	//case 0: // FROZEN.
+	//	zap.L().Warn("Updating (fake) database schema from 0 to 1...")
+	//	_, err = tx.Exec(`INSERT INTO ` + db.schema + `.migrations (schema_version) VALUES (1);`)
+	//	if err != nil {
+	//		return errors.Wrap(err, "sql.Tx.Exec (v0 -> v1)")
+	//	}
+	//	//fallthrough
+	//}
+
+	if err = tx.Commit(); err != nil {
+		return errors.Wrap(err, "sql.Tx.Commit")
+	}
+
+	return nil
+}
+
+func (db *postgresDatabase) executeTemplate(text string, data interface{}, funcs template.FuncMap) string {
```

Comment:
Feel free to remove this since it's unused.

Reply:
Ok. I'll probably remove it on this weekend when I'll be on my Linux machine.